tools/rosbag/package.xml

```diff
@@ -9,38 +13,39 @@
   <maintainer email="dthomas@osrfoundation.org">Dirk Thomas</maintainer>
   <license>BSD</license>
 
-  <url>http://ros.org/wiki/rosbag</url>
+  <url type="website">http://wiki.ros.org/rosbag</url>
+  <url type="bugtracker">https://github.com/ros/ros_comm/issues</url>
+  <url type="repository">https://github.com/ros/ros_comm</url>
   <author>Tim Field</author>
   <author>Jeremy Leibs</author>
   <author>James Bowman</author>
 
+  <depend>boost</depend>
+  <depend>rosbag_storage</depend>
+  <depend>rosconsole</depend>
+  <depend>roscpp</depend>
+  <depend>std_srvs</depend>
+  <depend>xmlrpcpp</depend>
+
   <buildtool_depend version_gte="0.5.78">catkin</buildtool_depend>
 
-  <build_depend>boost</build_depend>
   <build_depend>cpp_common</build_depend>
-  <build_depend>python-imaging</build_depend>
-  <build_depend>rosbag_storage</build_depend>
-  <build_depend>rosconsole</build_depend>
-  <build_depend>roscpp</build_depend>
+  <build_depend condition="$ROS_PYTHON_VERSION == 2">python-imaging</build_depend>
+  <build_depend condition="$ROS_PYTHON_VERSION == 3">python3-imaging</build_depend>
   <build_depend>roscpp_serialization</build_depend>
-  <build_depend>std_srvs</build_depend>
   <build_depend>topic_tools</build_depend>
-  <build_depend>xmlrpcpp</build_depend>
 
-  <run_depend>boost</run_depend>
-  <run_depend>genmsg</run_depend>
-  <run_depend>genpy</run_depend>
-  <run_depend>python-crypto</run_depend>
-  <run_depend>python-gnupg</run_depend>
-  <run_depend>python-rospkg</run_depend>
-  <run_depend>rosbag_storage</run_depend>
-  <run_depend>rosconsole</run_depend>
-  <run_depend>roscpp</run_depend>
-  <run_depend>roslib</run_depend>
-  <run_depend>rospy</run_depend>
-  <run_depend>std_srvs</run_depend>
-  <run_depend>topic_tools</run_depend>
-  <run_depend>xmlrpcpp</run_depend>
+  <exec_depend>genmsg</exec_depend>
+  <exec_depend>genpy</exec_depend>
+  <exec_depend condition="$ROS_PYTHON_VERSION == 2">python-crypto</exec_depend>
+  <exec_depend condition="$ROS_PYTHON_VERSION == 3">python3-crypto</exec_depend>
```

Comment:
There's no `python3-crypto` key, and the project is dead. [`python3-cryptography`](https://github.com/ros/rosdistro/blob/0ea30fed0bcb564b9e4d5b2d2efd8783e8e3afa9/rosdep/python.yaml#L4925-L4929) is a possible replacement #1599 https://github.com/ros/ros_comm/pull/1606

Reply:
Alternative PRs: #1609 or #1793. Currently I am leaning towards the former one due to the much simpler diff.

Reply:
With #1609 merged I rebased this patch and updated the dependency to `python(3)-pycryptodome`.